### PR TITLE
Addon unit tests

### PR DESCRIPTION
**Pull request by jenting**
## Why is this PR needed?

Unit tests for the addons to increase unit test coverage.

## What does this PR do?

Add unit tests

## Anything else a reviewer needs to know?

N/A

## Info for QA

This is info for QA so that they can validate this. This is **mandatory** if this PR fixes a bug.
If this is a new feature, a good description in "What does this PR do" may be enough.

### Related info

N/A

### Status **BEFORE** applying the patch

Package `internal/pkg/skuba/upgrade/addon` test coverage: 0%

### Status **AFTER** applying the patch

Package `internal/pkg/skuba/upgrade/addon` test coverage: 95%

## Docs

N/A

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

Signed-off-by: JenTing Hsiao <jenting.hsiao@suse.com>

### CODE REVIEW

File: internal/pkg/skuba/addons/addons.go [hunk 1]
```diff
@@ -75,9 +76,9 @@ const (
 	// executed inside a cluster definition folder
 	AlwaysRender ApplyBehavior = iota
 	// This is the desired behavior for Bootstrap, when the user can tweak
-	// the addon configurations after `skuba cluster init`, so
-	// `skuba node bootstrap` will apply the modified settings instead of
-	// re-rendering them forcefully
+	// the addon configurations aclientSetfter `skuba cluster init`, so
+	// `skuba node bootstrap` wilclientSetl apply the modified settings instead of
+	// re-rendering them forcefulclientSetly
```
Comment (ereslibre) [inline internal/pkg/skuba/addons/addons.go]: Some artifacts here :thinking:

File: internal/pkg/skuba/upgrade/addon/example_test.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+			kubernetes.Dex:     &kubernetes.AddonVersion{Version: "2.17.0", ManifestVersion: 1},
+			kubernetes.Gangway: &kubernetes.AddonVersion{Version: "3.1.0", ManifestVersion: 0},
+		},
+	})
+
+	// Output:
+	//   - cilium: 1.5.3 -> 1.5.3 (manifest version from 0 to 1)
+	//   - dex: 2.16.0 -> 2.17.0
+	//   - gangway: 3.1.0 (new addon)
+}
```
Comment (ereslibre) [inline internal/pkg/skuba/upgrade/addon/example_test.go]: This file looks like a leftover, right?

File: internal/pkg/skuba/upgrade/addon/versions_test.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+				},
+			},
+			expectedErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			got, gotErr := UpdatedAddons(tt.clientSet, tt.clusterVersion)
```
Comment (ereslibre) [inline internal/pkg/skuba/upgrade/addon/versions_test.go]: Ideally we can have helper functions so if we remove versions in the future from the map these tests don't need any adaptation.

Have a look at https://github.com/suse/skuba/blob/c7e9c95c047aa6915822f28a8a53df941af1a43c/internal/pkg/skuba/kubernetes/versions_test.go#L27-L67 on how we achieved this by using helper functions and some refactors; the tests could provide the available versions so we test the logic strictly without relying on the real `Versions` map.

File: internal/pkg/skuba/upgrade/addon/versions_test.go [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+		t.Run(tt.name, func(t *testing.T) {
+			got := HasAddonUpdate(tt.aviu)
+			if got != tt.expected {
+				t.Errorf("got: %v, expect: %v", got, tt.expected)
+			}
+		})
+	}
+}
+
+func ExamplePrintAddonUpdates() {
```
[the remark this answers is not in the document]
Comment (jenting) [reply internal/pkg/skuba/upgrade/addon/versions_test.go]: You could check this [blog](https://blog.golang.org/examples)
Golang example test will do full output comparison.

**Pull request merged**